### PR TITLE
docs: Updates Flutter SDK v1.0 docs with most recent code

### PR DESCRIPTION
**Pull request by dshukertjr**
> **Warning**
> Please don't merge this yet, as supabase-flutter v1.0 has not been released yet. 

We have not released the new supabase-flutter SDK yet, but we plan to launch it soon. I will merge this one once we release v1.0!

Also, this PR includes replacing various `Dart` with `Flutter`. This is to push the users to use supabase-flutter instead of supabase-dart for Flutter projects. Right now, because of the lack of this messaging, a lot of Flutter users are actually using supabase-dart, which lacks handling token refresh and few other features. With this update, I want to give a clear path to the Flutter users that they are supposed to use supabase-flutter.

### CODE REVIEW

**Comment by dshukertjr** [comment]
I want to replace the `dart` with `flutter` as well, but I would do that in a separate PR.

File: apps/reference/nav/supabase_dart_sidebars.js [hunk 1]
```diff
@@ -14,10 +14,14 @@ const sidebars = {
...
-        'generated/auth-signinwithprovider',
+        'generated/auth-signinwithpassword',
+        'generated/auth-signinwithotp',
+        'generated/auth-signinwithoauth',
         'generated/auth-signout',
-        'generated/auth-update',
+        'generated/auth-verifyotp',
+        'generated/auth-currentsession',
+        'generated/auth-currentuser',
+        'generated/auth-updateuser',
```
Comment (dannykng) [inline apps/reference/nav/supabase_dart_sidebars.js:24]: @dshukertjr are you seeing this issue when running the dev server locally?
```
[INFO] Starting the development server...
[ERROR] Loading of version failed for version current
[ERROR] Error: Invalid sidebar file at "nav/supabase_dart_sidebars.js".
These sidebar document ids do not exist:
- generated/auth-signinwithoauth
- generated/auth-signinwithotp
- generated/auth-signinwithpassword
- generated/auth-updateuser
- generated/auth-verifyotp
```

File: README.md [hunk 2]
```diff
@@ -119,8 +119,8 @@ Our approach for client libraries is modular. Each sub-library is a standalone i
     <td><a href="https://github.com/supabase-community/functions-csharp" target="_blank" rel="noopener noreferrer">functions-csharp</a></td>
   </tr>
   <tr>
-    <td>Dart (Flutter)</td>
-    <td><a href="https://github.com/supabase/supabase-dart" target="_blank" rel="noopener noreferrer">supabase-dart</a></td>
+    <td>Flutter</td>
+    <td><a href="https://github.com/supabase/supabase-flutter" target="_blank" rel="noopener noreferrer">supabase-flutter</a></td>
```
Comment (dannykng) [inline README.md:123]: Looks like there are a bunch of links that'll need to be updated with this change too. Seeing quite a few results for `git grep -i 'supabase-dart'`.

Comment (dshukertjr) [reply README.md:123]: Ah, yeah. I will replace those that are appropriate. Am I correct to say that I can ignore anything in the `apps/temp-docs`?

File: apps/reference/nav/supabase_dart_sidebars.js (hunk 1 above)
Comment (dshukertjr) [reply apps/reference/nav/supabase_dart_sidebars.js:24]: I couldn't reproduce it, but it seems like after fixing the broken links you referenced below, build is working.

File: README.md (hunk 2 above)
Comment (dshukertjr) [reply README.md:123]: Assuming I can ignore temp-docs, done!

**Review by kiwicopple — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (dannykng) [reply README.md:123]: Yep!

**Pull request merged**